pydm/main_window.py

```diff
@@ -498,3 +500,68 @@ def _confirm_quit(self, callback):
     def show_macro_window(self):
         macro_window = MacroWindow(self)
         macro_window.show()
+
+    def add_menu_items(self):
+        # create the custom menu with user given items
+        if not isinstance(self.display_widget(), Display):
+            return
+        items = self.display_widget().menu_items()
+        if len(items) != 0:
+            self.create_menu(self.ui.menuCustomActions, items)
+
+        # connect custom save, save as, and load functions
+        file_menu_items = self.display_widget().file_menu_items()
+        if len(file_menu_items) != 0:
+            valid_keys = ('save', 'save_as', 'load')
+            ui_actions = (self.ui.actionSave, self.ui.actionSave_As, self.ui.actionLoad)
+            action_dict = dict(zip(valid_keys, ui_actions))
+            # iterate through user given keys, which need to match the possible keys
+            for key in file_menu_items.keys():
+                fileAction = file_menu_items[key]
```

Comment:
Line not really needed since it's immediate redefined below. Could save a line and just iterate over key/value pairs
`for key, fileAction in file_menu_items.items():`
But not a big deal either way